test/units/testsuite-02.sh

```diff
@@ -34,77 +34,50 @@ fi
 # Arguments:
 #   $1: test path
 #   $2: test exit code
-report_result() {
-    if [[ $# -ne 2 ]]; then
-        echo >&2 "check_result: missing arguments"
+run_test() {
+    if [[ $# -ne 1 ]]; then
+        echo >&2 "run_test: missing arguments"
         exit 1
     fi
 
-    local name="${1##*/}"
-    local ret=$2
+    local test="$1"
+    local name="${test##*/}"
+
+    echo "Executing test $name as unit $name.service"
+
+    systemd-run --quiet --property Delegate=1 --unit="$name" --wait "$test" && ret=0 || ret=$?
+
+    exec 420</lock
+    flock 420
```

Comment:
The `{varname}` redirection form can be used to allocate an available fd without having to guess at a high free fd.

```suggestion
    exec {LOCK_FD}</lock
    flock "$LOCK_FD"
```